inspect/inspect.go

```diff
@@ -107,6 +107,10 @@ func (ins *Inspector) Run(ctx context.Context) error {
 	return startRPCServers(ctx, ins.config, ins.logger, ins.routes)
 }
 
+// SetLogger allows callers to inject a logger into the inspector
+// instance.
+func (ins *Inspector) SetLogger(logger log.Logger) { ins.logger = logger }
```

Comment:
I would like to put in a plug for a more principled change to the constructor API, since we're in here anyway. Rather than exporting methods to poke logging and other concerns into the object after it's constructed, I recommend using an options type at construction time.

It's a little more work to set up, but easier (IMO) to maintain. There's more than one way to do that, the one I find simplest is https://play.golang.org/p/UEuwewv7Qx6. (I put some explanatory comments in the example)

That said: I _do_ like the idea of getting some of the constructor args off the positional mainline, on basic legibility grounds at the call site.

Reply:
Sure, I move the logger to an argument. :D 

I think deep down "inspect" is a "mode" for running a tendermint node, and in an ideal world, we might imagine that it's part of "node" in some way. At the same time, this isn't the right way to factor the code at the moment (because `node` is a fire.) Which is just to say that a lot of the flaws here are probably known, and hopefully transient.

Reply:
I don't think it lands anywhere really on the roadmap, but I'd prefer we modularize our code in such a way so that writing something like the 'inspect' code is reasonably trivial given the libraries we have. It was reasonably straightforward, but the `rpc` packages make it a bit convoluted.

Reply:
For folks outside the core, the inspector is a mode. But for us it's a regular old library, and I'd ideally like not to have things be too deeply weird and/or defensive against that use. 

We want to be able to (re)use, test, (re)wire libraries in various ways, especially as we're moving things around (e.g., to clean up other weirdnesses 🙂) so IMO we should build them with the same kind of interface we would expose if it WERE used by outsiders.  The `/internal/` designation protects us sufficiently, I think, from outside consumption.